src/test/scala/mesosphere/marathon/storage/migration/MigrationTo18Test.scala

```diff
@@ -0,0 +1,135 @@
+package mesosphere.marathon
+package storage.migration
+
+import akka.Done
+import akka.stream.scaladsl.Source
+import com.typesafe.scalalogging.StrictLogging
+import mesosphere.AkkaUnitTest
+import mesosphere.marathon.core.condition.Condition
+import mesosphere.marathon.core.instance.Instance
+import mesosphere.marathon.core.storage.store.impl.zk.ZkPersistenceStore
+import mesosphere.marathon.core.task.Task
+import mesosphere.marathon.core.task.state.NetworkInfo
+import mesosphere.marathon.state.{PathId, Timestamp}
+import mesosphere.marathon.storage.repository.InstanceRepository
+import org.apache.mesos
+import org.apache.mesos.Protos.NetworkInfo.Protocol
+import org.scalatest.Inspectors
+import play.api.libs.json.{JsObject, JsString, JsValue, Json}
+
+import scala.concurrent.Future
+
+class MigrationTo18Test extends AkkaUnitTest with StrictLogging with Inspectors {
+
+  "Migration to 18" should {
+    "Update instance condition from Created to Provisioned" in {
+
+      Given("two instances in Created state")
+      val f = new Fixture()
+
+      val instanceId1 = Instance.Id.forRunSpec(PathId("/app"))
+      val instanceId2 = Instance.Id.forRunSpec(PathId("/app2"))
+      val instanceId3 = Instance.Id.forRunSpec(PathId("/app3"))
+
+      val instance1Json = f.created(instanceId1)
+      val instance1 = instance1Json.as[Instance]
+      val migratedInstance1 = f.setProvisionedCondition(instance1)
+
+      val instance2Json = f.created(instanceId2)
+      val instance2 = instance2Json.as[Instance]
+      val migratedInstance2 = f.setProvisionedCondition(instance2)
+
+      val instance3Json = f.provisioned(instanceId3)
+      val instance3 = instance3Json.as[Instance]
+
+      f.instanceRepository.ids() returns Source(List(instanceId1, instanceId2, instanceId3))
+      f.persistenceStore.get[Instance.Id, JsValue](equalTo(instanceId1))(any, any) returns Future(Some(instance1Json))
+      f.persistenceStore.get[Instance.Id, JsValue](equalTo(instanceId2))(any, any) returns Future(Some(instance2Json))
+      f.persistenceStore.get[Instance.Id, JsValue](equalTo(instanceId3))(any, any) returns Future(Some(instance3Json))
+      f.instanceRepository.store(equalTo(migratedInstance1)) returns Future.successful(Done)
+      f.instanceRepository.store(equalTo(migratedInstance2)) returns Future.successful(Done)
+      f.instanceRepository.store(equalTo(instance3)) returns Future.successful(Done) //instance is unchanged
+
+      When("they are migrated")
+      MigrationTo18.migrateInstanceConditions(f.instanceRepository, f.persistenceStore).futureValue
```

Comment:
Why not simple run a stream though the flow. I don't think we need the extra mocks. See [MigrationTo17Test](https://github.com/mesosphere/marathon/blob/master/src/test/scala/mesosphere/marathon/storage/migration/MigrationTo17Test.scala#L44) for an example.